app/lib/multiple_parameters_date_type.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class MultipleParametersDateType < ActiveModel::Type::Value
+  class PotentialDate
+    include ActiveModel::Model
+    attr_accessor :year, :month, :day
+  end
+
+  def self.process(params)
+    params.keys.select { |key| key.to_s.match?(/\(\d+i\)$/) }.map do |key|
+      attribute_name = key.to_s.split('(').first
+
+      year = params.delete("#{attribute_name}(1i)")
+      month = params.delete("#{attribute_name}(2i)")
+      day = params.delete("#{attribute_name}(3i)")
```

Comment:
In the future we might need to hold this three fields in an attr_accessor and alias with 3i,2i,1i but it not needed for the form on this PR. I hope this evolves a little with other forms using it.